### DISCUSSION TITLE
LOWER MESH REFINEMENT AT EB

### DISCUSSION THREAD

**Discussion by PatriciaIG** [AMReX-Combustion/PeleC#752] [Q&A] [answered]
Hi, everyone,
Is it possible to have a lower amr level at the EB than the global amr? I want to refine a two-level at the borders of the EB and 4 levels where the temperature is over a value, lotemperr.  Thank you in advance for your help.

**Comment by baperry2** [comment, marked as answer]
Yes, this is possible even though by default the EB is entirely refined to the global amr max level. 

- You can for example set `tagging.eb_refine_type = "static"` and `tagging.max_eb_refine_lev = 2`, to statically refine the EB to a maximum of two levels (in this setup, if other refinement criteria exceed `max_eb_refine_lev` at the EB surface, they are ignored at the EB surface such that the entire EB surface is always at refined to exactly `max_eb_refine_lev`).
- Alternatively, you could set `tagging.eb_refine_type = "adaptive"` and `tagging.min_eb_refine_lev = 2`, which refines the EB surface to 2 levels of refinement, and allows additional adaptive refinement that crosses the EB surface based on other refinement indicators. Coarse-fine interfaces across the EB surface are then handled through a process called [re-redistribution](https://amrex-combustion.github.io/PeleC/geometry/EB.html#re-redistribution)

**Reply by FernandoIGuerra** [reply]
i am encountering some problems using the "adaptive" feature. No matter the min_eb_level that i use, PeleC still set the EB as the amr.max_level. This does not happen with the static one. The inputs are as follow:

tagging.denratio = 1.1
tagging.max_denratio_lev =  2

pelec.flame_trac_name = HO2
tagging.max_ftracerr_lev = 2
tagging.ftracerr = 0.0002

tagging.denerr = 0.001
tagging.max_denerr_lev = 2

tagging.eb_refine_type = "adaptive"
tagging.min_eb_refine_lev = 1

Thanks in advance!!

**Reply by baperry2** [reply]
So apparently there's a redundant tagging option that tags any cells with vfrac between 0 and 1, which by default will tag all EB boundary cells. You can turn that off with `tagging.max_vfracerr_lev = 0` and then adaptive tagging should work as anticipated.
